feature: new rule inefficient-map-lookup (#1491)
mgechev/revive: RULES_DESCRIPTIONS.md

arguments = ["preserve-scope"]
```

## inefficient-map-lookup

_Description_: This rule spots code that iteratively search for a map key.

Example:

> ```golang

Reviewer: I extended docs #1494 about `golang`.

arguments = ["preserve-scope"]
```

## inefficient-map-lookup

> _Description_: This rule spots code that iteratively search for a map key.

Reviewer: ```suggestion
_Description_: This rule identifies code that iteratively searches for a key in a map.
```

> This inefficiency is usually introduced when refactoring from a slice to a map.

Reviewer: ```suggestion
This inefficiency is usually introduced when refactoring code from using a slice to a map.
```

For example if during refactoring the `elements` slice is transformed into a map.

```diff
-       elements             []string
+       elements             map[string]float64
```

> and then a loop across `elements` is changed in an obvious but inefficient way:

Reviewer: ```suggestion
and then a loop over `elements` is changed in an obvious but inefficient way:
```

```diff
-       for _, e := range elements {
+       for e := range elements {
                if e == someStaticValue {
                        // do something
                }
        }
```

Example:

```go
aMap := map[string]bool{}{}
aValue := false

// Inefficient map lookup
for k := range aMap {
  if k == aValue {
    // do something
  }
}

> // Simpler and efficient version

Reviewer: ```suggestion
// Simpler and more efficient version
```
